pkg/kv/kvserver/store_raft.go

```diff
@@ -139,7 +148,24 @@ func (qs *raftReceiveQueues) LoadOrCreate(
 	}
 	q := &raftReceiveQueue{maxLen: maxLen}
 	q.acc.Init(context.Background(), qs.mon)
-	return qs.m.LoadOrStore(rangeID, q)
+	q, loaded = qs.m.LoadOrStore(rangeID, q)
+	if !loaded {
+		// The sampling of enforceMaxLen can be concurrent with a call to
+		// SetEnforceMaxLen. We can sample a stale value, then SetEnforceMaxLen
+		// can fully execute, and then set the stale value here. Since
+		// SetEnforceMaxLen sets the latest value first, before iterating over the
+		// map, it suffices to check after setting the value here that it has not
+		// changed.
+		for {
+			enforceBefore := qs.enforceMaxLen.Load()
+			q.SetEnforceMaxLen(enforceBefore)
+			enforceAfter := qs.enforceMaxLen.Load()
+			if enforceAfter == enforceBefore {
+				break
+			}
+		}
```

Comment:
Worth explaining why this is sufficient. After the second `qs.enforceMaxLen.Load()`, we know that `q.SetEnforceMaxLen` has been already called (by us and maybe also by `qs.SetEnforceMaxLen`), or will be called soon by `qs.SetEnforceMaxLen`.

Is this resilient to ABA problem? Seems like yes. Even if `enforceMaxLen` has been changes multiple times in between two `Load`s and the second `Load` observes a different incarnation of the same value, the latest value will be applied by `qs.SetEnforceMaxLen`.

Reply:
Added more to the comment.
<!-- Reviewable comment -ODnCuDIBrkwB057ruxX:bjkivct -->
<!-- Sent from Reviewable.io -->